pkg/api/validation/validation.go

```diff
@@ -128,7 +128,7 @@ func ValidateNodeName(name string, prefix bool) (bool, string) {
 // Prefix indicates this name will be used as part of generation, in which case
 // trailing dashes are allowed.
 func ValidateNamespaceName(name string, prefix bool) (bool, string) {
-	return nameIsDNSSubdomain(name, prefix)
+	return nameIsDNS952Label(name, prefix)
```

Comment:
Should be IsDNS1123Label() I think - no reason for the more restrictive format that I can see.